refactor: using ibis.expr.api.schema inside ibis.expr.api.table

EdAbati opened the pull request:
fixes #3040

I had to create an alias for the `schema` function since the `table` function has a `schema` argument, please let me know if there is a cleaner approach or if I missed something.
Thank you very much!

cpcloud commented on ibis/expr/api.py:
```diff
@@ -283,11 +286,16 @@ def schema(
     ...              types=['string', 'int64', 'boolean'])
     """
     if pairs is not None:
+        if isinstance(pairs, dict):
+            return Schema.from_dict(pairs)
         return Schema.from_tuples(pairs)
```
```suggestion
        return Schema.from_dict(dict(pairs))
```

cpcloud commented on ibis/expr/api.py:
```diff
@@ -258,7 +258,9 @@ def sequence(values):
 def schema(
-    pairs: Iterable[tuple[str, dt.DataType]] | None = None,
+    pairs: Iterable[tuple[str, dt.DataType]]
+    | dict[str, dt.DataType]
```
```suggestion
    | Mapping[str, dt.DataType]
```

cpcloud reviewed (changes requested):
Thanks @EdAbati! Couple of small suggestions then this is good to go!

pep8speaks commented:
Hello @EdAbati! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`ibis/expr/api.py`](https://github.com/ibis-project/ibis/blob/42e11fa07c22a33d7f10713a83ea330856be6769/ibis/expr/api.py):

> [Line 262:7](https://github.com/ibis-project/ibis/blob/42e11fa07c22a33d7f10713a83ea330856be6769/ibis/expr/api.py#L262): [F821](https://duckduckgo.com/?q=pep8%20F821) undefined name 'Mapping'

EdAbati commented:
Hi @cpcloud , I removed the `Schema.from_tuples` as you suggested but some tests failed. It seems that `Schema.from_dict` didn't support empty schemas.

I fixed it in the last commit.

cpcloud commented on ibis/expr/api.py:
```diff
@@ -283,11 +286,14 @@ def schema(
         return Schema(names, types)
 
 
+_schema_func = schema
```
```suggestion
_schema = schema
```

cpcloud commented on ibis/expr/api.py:
```diff
@@ -306,10 +312,7 @@ def table(schema, name=None):
-            schema = Schema.from_dict(schema)
-        else:
-            schema = Schema.from_tuples(schema)
+        schema = _schema_func(pairs=schema)
```
```suggestion
        schema = _schema(pairs=schema)
```

cpcloud reviewed (changes requested):
Small change from `_schema_func -> _schema` then this is good to go! Thanks for digging into this!

EdAbati commented:
Sorry I had to revert the last commit and force push. I used `cz` and named the commit `refactor: renamed _schema_func to _schema`. After I pushed, I had only the commit in my fork but it wasn't showing in this PR.
I am not super familiar with `cz`, did I make a mistake? Should it just be used for the message of the 1st commit of a branch?

cpcloud commented:
@EdAbati No worries!

`cz` doesn't need be used for every commit. However, in the end you'll need to squash to 1 commit and it needs to be used for that one.

With that in mind, would you mind squashing and fixing up the final commit? Will land after that!

EdAbati commented:
Hi @cpcloud , after a few trials I now squashed everything into 1 commit. Let me know if I missed something. Thanks

cpcloud reviewed (approved):
Awesome! Thank @EdAbati!
